contracts/destChain/Prover-Axelar.sol

```diff
@@ -52,10 +53,8 @@ contract DealClientAxl is AxelarExecutable {
     mapping(bytes => uint256) public providerGasFunds; // Funds set aside for calling oracle by provider
     mapping(uint256 => DestinationChain) public chainIdToDestinationChain;
 
-    constructor(
-        address _gateway,
-        address _gasReceiver
-    ) AxelarExecutable(_gateway) {
+    function initialize(address _gateway, address _gasReceiver) public initializer{
+        __AxelarExecutable_init(_gateway);
         gasService = IAxelarGasService(_gasReceiver);
     }
 
```

Comment:
Maybe better to reserve storage gap for any potential upgraded adding new variables.